flink-runtime/src/main/java/org/apache/flink/runtime/state/memory/AbstractMemStateSnapshot.java

```diff
@@ -54,6 +56,8 @@
 
 	/** The serialized data of the state key/value pairs */
 	private final byte[] data;
+	
+	private transient boolean closed;
```

Comment:
I think it is not crucial to have a strict barrier here. If the reading thread eventually notices the flag, it is enough. And since volatile accesses are much more expensive, I wanted to avoid that.